packages/core/src/App/Containers/ResetEmailModal/__tests__/reset-email-modal.spec.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import ResetEmailModal from '../reset-email-modal';
+import { ConfirmEmailModal } from '../../ConfirmEmailModal/confirm-email-modal';
+
+jest.mock('../../ConfirmEmailModal/confirm-email-modal', () => ({
+    ConfirmEmailModal: jest.fn(({ is_open, onClose }: Partial<React.ComponentProps<typeof ConfirmEmailModal>>) =>
+        is_open ? (
+            <div>
+                <p>ConfirmEmailModal</p>
+                <button onClick={onClose}>Close</button>
+            </div>
+        ) : null
+    ),
+}));
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    getErrorMessages: jest.fn(() => ({
+        email: jest.fn(() => 'Invalid email address.'),
+    })),
+}));
+
+describe('ResetEmailModal', () => {
+    const mock_store = mockStore({
+        ui: {
+            is_reset_email_modal_visible: true,
+        },
+        client: {
+            email: 'test@example.com',
+        },
+    });
+    const wrapper = ({ children }: { children: JSX.Element }) => (
+        <StoreProvider store={mock_store}>{children};</StoreProvider>
+    );
+
+    const renderComponent = () => {
+        render(<ResetEmailModal />, { wrapper });
+    };
+
+    it('renders the reset email modal', () => {
+        renderComponent();
+
+        expect(screen.getByText(/Enter a new email address/i)).toBeInTheDocument();
+        expect(screen.getByPlaceholderText(/Email address/i)).toBeInTheDocument();
+        expect(screen.queryByText('ConfirmEmailModal')).not.toBeInTheDocument();
```

Comment:
I see these 2 key words are being duplicated maybe we can define a constant value for it